shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2462,6 +2466,23 @@ - (void)clearTextInputClient {
   _activeView.frame = CGRectZero;
 }
 
+- (void)updateConfig:(NSDictionary*)dictionary {
+  BOOL isSecureTextEntry = [dictionary[kSecureTextEntry] boolValue];
+  for (UIView* view in self.textInputViews) {
+    if ([view isKindOfClass:[FlutterTextInputView class]]) {
+      FlutterTextInputView* inputView = (FlutterTextInputView*)view;
+      // The feature of holding and draging spacebar to move cursor is affected by
+      // secureTextEntry, so when obscureText is updated, we need to update secureTextEntry
+      // and call reloadInputViews.
+      // https://github.com/flutter/flutter/issues/122139
```

Comment:
not sure if i follow this. can you explain more why setting secrueTextEntry and the reload fixes the space bar cursor interaction? Or is it an undocumented behavior by apple here?

Reply:
First of all, it is clear that when `secureTextEntry` is true, the space bar cursor interaction is not allowed, otherwise it is allowed. This is the iOS native correct interaction behavior. So we need to update `secureTextEntry` in time when switching `obscureText`. I did not find a description of whether this space bar cursor interaction is affected by `secureTextEntry` in Apple's official documentation, this is just the conclusion obtained after testing.

Then, when the value of the `secureTextEntry` is changed, the interaction and style of the keyboard are still before the `secureTextEntry` is changed, so here I call the `reloadInputViews` method to refresh keyboard.